test/functionalTest/cases/2553_wrong_metadata_type_in_prev_value_in_notification/wrong_metadata_type_in_prev_value_in_notification.test

```diff
@@ -0,0 +1,193 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Wrong metadata type in prevValue in notification
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1/A1
+# 02. Create E1/A1 with type: a1
+# 03. See notification and reset
+# 04. Update A1 using PUT /v2/entities/E1/attrs/A1/value
+# 05. See notification and make sure the type of E1/A1 is a1
+#
+
+echo "01. Create subscription for E1/A1"
+echo "================================="
+payload='{
+  "notification": {
+    "http": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "metadata": [ "previousValue" ]
+  },
+  "subject": {
+    "entities": [
+      {
+        "idPattern": ".*"
+      }
+    ]
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create E1/A1 with type: a1"
+echo "=============================="
+payload='{
+  "id": "E1",
+  "A1": {
+    "value": 1,
+    "type": "a1"
```

Comment:
The type equal to the attribute name... a bit confusing. I'd suggest to use T1 for the type.